staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -88,8 +94,6 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	timeoutCh := r.Context().Done()
```

Comment:
@tkashem you were working on fixing this, IIRC you already had a PR for removing the timeout logic using goroutines

Reply:
I'm all for removing the goroutine. It feels like a pretty big hack.

Reply:
That PR seems to be https://github.com/kubernetes/kubernetes/pull/114189.

But it doesn't remove this extra goroutine in timeout.go (yet?). Is that planned once the per-request timeouts work?

Reply:
@pohly @aojea  https://github.com/kubernetes/kubernetes/pull/114189 is the first step, and then the follow up step will be removing the timeout handler.

Reply:
also, can you please share with me the stack trace for the timeout handler data race? (I could not find it in the comments)